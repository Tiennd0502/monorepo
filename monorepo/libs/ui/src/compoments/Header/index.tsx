import { ReactNode, memo } from 'react';

import { YStack, XStack } from 'tamagui';
import Heading from '../Heading';

interface HeaderProps {
  startIcon?: ReactNode;
  title?: string;
  subTitle?: string;
  endIcon?: ReactNode;
}

const Header = ({ startIcon, title, subTitle, endIcon }: HeaderProps) => {
  return (
    <XStack justifyContent="space-between" alignItems="center">
      <YStack width="$10">{startIcon}</YStack>
      <YStack alignItems="center">
        {subTitle && (
          <Heading fontWeight="600" color="$textTertiary">
            {subTitle}
          </Heading>
        )}
        {title && <Heading color="$primary">{title}</Heading>}
      </YStack>
      <YStack width="$10">{endIcon}</YStack>
    </XStack>
  );
};

export default memo(Header);
