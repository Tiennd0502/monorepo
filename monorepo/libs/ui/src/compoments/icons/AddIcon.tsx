import Svg, { Path, SvgProps, Mask } from 'react-native-svg';

import { TOKEN_KEYS } from '@monorepo/types';
import { getTokenValueByKey } from '@monorepo/utils';

export const AddIcon = ({
  width = 24,
  height = 25,
  color = '$primary',
  ...props
}: SvgProps) => (
  <Svg
    testID='add-icon'
    fill="none"
    viewBox="0 0 24 25"
    width={getTokenValueByKey(TOKEN_KEYS.SPACE, width.toString())}
    height={getTokenValueByKey(TOKEN_KEYS.SPACE, height.toString())}
    {...props}
  >
    <Path
      d="M19 11.5h-6v-6a1 1 0 10-2 0v6H5a1 1 0 100 2h6v6a1 1 0 102 0v-6h6a1 1 0 100-2z"
      fill={getTokenValueByKey(TOKEN_KEYS.COLOR, color.toString())}
    />
    <Mask maskUnits="userSpaceOnUse" x={4} y={4} width={16} height={17}>
      <Path
        d="M19 11.5h-6v-6a1 1 0 10-2 0v6H5a1 1 0 100 2h6v6a1 1 0 102 0v-6h6a1 1 0 100-2z"
        fill="#fff"
      />
    </Mask>
  </Svg>
);
