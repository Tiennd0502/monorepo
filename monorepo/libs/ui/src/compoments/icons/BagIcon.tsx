import Svg, { Path, SvgProps } from 'react-native-svg';

import { TOKEN_KEYS } from '@monorepo/types';
import { getTokenValueByKey } from '@monorepo/utils';

export const BagIcon = ({
  width = 20,
  height = 20,
  color = '$primary',
  ...props
}: SvgProps) => (
  <Svg
    fill="none"
    viewBox="0 0 20 20"
    width={getTokenValueByKey(TOKEN_KEYS.SPACE, width.toString())}
    height={getTokenValueByKey(TOKEN_KEYS.SPACE, height.toString())}
    {...props}
  >
    <Path
      d="M17.969 16.31L16.667 4.06c0-.77-.597-1.393-1.334-1.393H4.667c-.737 0-1.334.623-1.334 1.394L2.031 16.31c-.019.095-.031.193-.031.295C2 17.376 2.597 18 3.333 18h13.334c.736 0 1.333-.624 1.333-1.394 0-.102-.012-.2-.031-.295zM13.333 6.088v.762c0 1.921-1.495 3.484-3.333 3.484S6.667 8.77 6.667 6.85v-.762A1.04 1.04 0 016 5.107c0-.578.448-1.046 1-1.046s1 .468 1 1.045V6.85c0 1.152.897 2.09 2 2.09s2-.938 2-2.09V5.106c0-.577.448-1.045 1-1.045s1 .468 1 1.045c0 .455-.28.838-.667.981z"
      fill={getTokenValueByKey(TOKEN_KEYS.COLOR, color.toString())}
    />
  </Svg>
);
