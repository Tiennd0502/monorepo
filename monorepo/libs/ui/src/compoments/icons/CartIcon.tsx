import Svg, { Path, SvgProps } from 'react-native-svg';

import { TOKEN_KEYS } from '@monorepo/types';
import { getTokenValueByKey } from '@monorepo/utils';

export const CartIcon = ({
  width = 24,
  height = 25,
  color = '$primary',
  ...props
}: SvgProps) => (
  <Svg
    fill="none"
    viewBox="0 0 24 25"
    width={getTokenValueByKey(TOKEN_KEYS.SPACE, width.toString())}
    height={getTokenValueByKey(TOKEN_KEYS.SPACE, height.toString())}
    {...props}
  >
    <Path
      fillRule="evenodd"
      clipRule="evenodd"
      d="M2.607 4.643A.643.643 0 013.25 4h1.928a.643.643 0 01.624.487l.52 2.084H21.25a.643.643 0 01.623.799l-1.928 7.714a.643.643 0 01-.624.487H7.75a.643.643 0 01-.624-.487l-2.45-9.798H3.25a.643.643 0 01-.643-.643zm4.037 3.214l1.607 6.429H18.82l1.607-6.429H6.644zm2.392 10.286a1.286 1.286 0 100 2.571 1.286 1.286 0 000-2.571zm-2.572 1.286a2.571 2.571 0 115.143 0 2.571 2.571 0 01-5.143 0zm11.572-1.286a1.286 1.286 0 100 2.571 1.286 1.286 0 000-2.571zm-2.572 1.286a2.571 2.571 0 115.143 0 2.571 2.571 0 01-5.143 0z"
      fill={getTokenValueByKey(TOKEN_KEYS.COLOR, color.toString())}
    />
  </Svg>
);
