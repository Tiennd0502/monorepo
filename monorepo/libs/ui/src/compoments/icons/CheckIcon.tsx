import Svg, { Path } from 'react-native-svg';

import { IconProps } from '@monorepo/types';

export const CheckIcon = ({
  width = 15,
  height = 12,
  color = '#fff',
}: IconProps) => {
  return (
    <Svg width={width} height={height} viewBox="0 0 15 12" fill="none">
      <Path
        d="M4.767 9.468L1.21 5.736 0 6.998 4.767 12 15 1.262 13.798 0 4.767 9.468z"
        fill={color}
      />
    </Svg>
  );
};
