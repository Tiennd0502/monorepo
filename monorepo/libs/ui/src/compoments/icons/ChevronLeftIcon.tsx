import Svg, { Path, SvgProps } from 'react-native-svg';

import { TOKEN_KEYS } from '@monorepo/types';
import { getTokenValueByKey } from '@monorepo/utils';

export const ChevronLeftIcon = ({
  width = 20,
  height = 20,
  color = '$primary',
  ...props
}: SvgProps) => (
  <Svg
    fill="none"
    viewBox="0 0 20 20"
    width={getTokenValueByKey(TOKEN_KEYS.SPACE, width.toString())}
    height={getTokenValueByKey(TOKEN_KEYS.SPACE, height.toString())}
    {...props}
  >
    <Path
      d="M13.25 4l-6.5 6.5 6.5 6.5"
      stroke={getTokenValueByKey(TOKEN_KEYS.COLOR, color.toString())}
      strokeWidth={1.5}
      strokeLinecap="round"
      strokeLinejoin="round"
    />
  </Svg>
);
