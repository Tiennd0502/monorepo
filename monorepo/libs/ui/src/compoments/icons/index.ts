export * from './ShowIcon';
export * from './LogoIcon';
export * from './HomeIcon';
export * from './StarIcon';
export * from './CheckIcon';
export * from './HiddenIcon';
export * from './ProfileIcon';
export * from './FavoriteIcon';
export * from './NotificationIcon';
