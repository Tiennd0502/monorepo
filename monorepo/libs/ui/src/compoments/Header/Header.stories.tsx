import { Meta, StoryObj } from '@storybook/react';

import Header from '.';
import { CartIcon, ChevronLeftIcon } from '../icons';

const meta: Meta<typeof Header> = {
  title: 'Components/Header',
  component: Header,
};

export default meta;

type Story = StoryObj<typeof Header>;

const Render = () => ({
  subTitle: 'Make home',
  title: 'BEAUTIFUL',
  startIcon: <ChevronLeftIcon />,
  endIcon: <CartIcon />,
});

export const Default: Story = {
  args: {
    subTitle: 'Make home',
    title: 'BEAUTIFUL',
    startIcon: <ChevronLeftIcon />,
    endIcon: <CartIcon />,
  },
};

export const OnlyTitle: Story = {
  args: {
    title: 'Profile',
  },
};

export const HasStartIcon: Story = {
  args: {
    title: 'BEAUTIFUL',
    startIcon: <ChevronLeftIcon />,
  },
};
