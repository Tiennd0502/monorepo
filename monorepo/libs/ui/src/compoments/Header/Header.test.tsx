import {render} from '@testing-library/react-native';

import { TamaguiProvider } from '@monorepo/provider';
import { CartIcon, ChevronLeftIcon } from '../icons';
import Header from '.';

const baseProps = {
  title: 'BEAUTIFUL',
};

const setup = (moreProps = {}) => {
  const props = {
    ...baseProps,
    ...moreProps,
  };
  
  return render(
    <TamaguiProvider>
      <Header {...props} />
    </TamaguiProvider>
  );
};


describe('Header', () => {
  it('should render Header component', () => {
    const container = setup({
      subTitle: 'Make home',
      endIcon: <CartIcon />,
    });

    expect(container).toMatchSnapshot();
  });

  it('should render Header component with only title prop', () => {
    const container = setup();

    expect(container).toMatchSnapshot();
  });

  it('should render Header component with not end icon', () => {
    const container = setup({
      startIcon: <ChevronLeftIcon />,
      endIcon: <CartIcon />,
    });

    expect(container).toMatchSnapshot();
  });
});
