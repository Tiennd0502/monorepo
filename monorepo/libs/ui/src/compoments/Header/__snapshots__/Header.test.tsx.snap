// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render Header component 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "space-between",
    }
  }
>
  <View
    style={
      {
        "flexDirection": "column",
        "width": 40,
      }
    }
  />
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "column",
      }
    }
  >
    <Text
      accessibilityRole="header"
      allowFontScaling={false}
      selectable={true}
      style={
        {
          "color": "#909090",
          "fontFamily": "Merriweather, Helvetica, Arial, sans-serif",
          "fontSize": 18,
          "lineHeight": 28,
          "marginBottom": 0,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 0,
        }
      }
      suppressHighlighting={true}
    >
      Make home
    </Text>
    <Text
      accessibilityRole="header"
      allowFontScaling={false}
      selectable={true}
      style={
        {
          "color": "#242424",
          "fontFamily": "Merriweather-Bold",
          "fontSize": 18,
          "lineHeight": 28,
          "marginBottom": 0,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 0,
        }
      }
      suppressHighlighting={true}
    >
      BEAUTIFUL
    </Text>
  </View>
  <View
    style={
      {
        "flexDirection": "column",
        "width": 40,
      }
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="25"
      bbWidth="24"
      fill="none"
      focusable={false}
      height="25"
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "flex": 0,
            "height": 25,
            "width": 24,
          },
        ]
      }
      vbHeight={25}
      vbWidth={24}
      width="24"
    >
      <RNSVGGroup
        fill={null}
        propList={
          [
            "fill",
          ]
        }
      >
        <RNSVGPath
          clipRule={0}
          d="M2.607 4.643A.643.643 0 013.25 4h1.928a.643.643 0 01.624.487l.52 2.084H21.25a.643.643 0 01.623.799l-1.928 7.714a.643.643 0 01-.624.487H7.75a.643.643 0 01-.624-.487l-2.45-9.798H3.25a.643.643 0 01-.643-.643zm4.037 3.214l1.607 6.429H18.82l1.607-6.429H6.644zm2.392 10.286a1.286 1.286 0 100 2.571 1.286 1.286 0 000-2.571zm-2.572 1.286a2.571 2.571 0 115.143 0 2.571 2.571 0 01-5.143 0zm11.572-1.286a1.286 1.286 0 100 2.571 1.286 1.286 0 000-2.571zm-2.572 1.286a2.571 2.571 0 115.143 0 2.571 2.571 0 01-5.143 0z"
          fill={
            {
              "payload": 4280558628,
              "type": 0,
            }
          }
          fillRule={0}
          propList={
            [
              "fill",
              "fillRule",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</View>
`;

exports[`Header should render Header component with not end icon 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "space-between",
    }
  }
>
  <View
    style={
      {
        "flexDirection": "column",
        "width": 40,
      }
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="20"
      bbWidth="20"
      fill="none"
      focusable={false}
      height="20"
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "flex": 0,
            "height": 20,
            "width": 20,
          },
        ]
      }
      vbHeight={20}
      vbWidth={20}
      width="20"
    >
      <RNSVGGroup
        fill={null}
        propList={
          [
            "fill",
          ]
        }
      >
        <RNSVGPath
          d="M13.25 4l-6.5 6.5 6.5 6.5"
          fill={
            {
              "payload": 4278190080,
              "type": 0,
            }
          }
          propList={
            [
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={
            {
              "payload": 4280558628,
              "type": 0,
            }
          }
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth="1.5"
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "column",
      }
    }
  >
    <Text
      accessibilityRole="header"
      allowFontScaling={false}
      selectable={true}
      style={
        {
          "color": "#242424",
          "fontFamily": "Merriweather-Bold",
          "fontSize": 18,
          "lineHeight": 28,
          "marginBottom": 0,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 0,
        }
      }
      suppressHighlighting={true}
    >
      BEAUTIFUL
    </Text>
  </View>
  <View
    style={
      {
        "flexDirection": "column",
        "width": 40,
      }
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="25"
      bbWidth="24"
      fill="none"
      focusable={false}
      height="25"
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "flex": 0,
            "height": 25,
            "width": 24,
          },
        ]
      }
      vbHeight={25}
      vbWidth={24}
      width="24"
    >
      <RNSVGGroup
        fill={null}
        propList={
          [
            "fill",
          ]
        }
      >
        <RNSVGPath
          clipRule={0}
          d="M2.607 4.643A.643.643 0 013.25 4h1.928a.643.643 0 01.624.487l.52 2.084H21.25a.643.643 0 01.623.799l-1.928 7.714a.643.643 0 01-.624.487H7.75a.643.643 0 01-.624-.487l-2.45-9.798H3.25a.643.643 0 01-.643-.643zm4.037 3.214l1.607 6.429H18.82l1.607-6.429H6.644zm2.392 10.286a1.286 1.286 0 100 2.571 1.286 1.286 0 000-2.571zm-2.572 1.286a2.571 2.571 0 115.143 0 2.571 2.571 0 01-5.143 0zm11.572-1.286a1.286 1.286 0 100 2.571 1.286 1.286 0 000-2.571zm-2.572 1.286a2.571 2.571 0 115.143 0 2.571 2.571 0 01-5.143 0z"
          fill={
            {
              "payload": 4280558628,
              "type": 0,
            }
          }
          fillRule={0}
          propList={
            [
              "fill",
              "fillRule",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</View>
`;

exports[`Header should render Header component with only title prop 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "space-between",
    }
  }
>
  <View
    style={
      {
        "flexDirection": "column",
        "width": 40,
      }
    }
  />
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "column",
      }
    }
  >
    <Text
      accessibilityRole="header"
      allowFontScaling={false}
      selectable={true}
      style={
        {
          "color": "#242424",
          "fontFamily": "Merriweather-Bold",
          "fontSize": 18,
          "lineHeight": 28,
          "marginBottom": 0,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 0,
        }
      }
      suppressHighlighting={true}
    >
      BEAUTIFUL
    </Text>
  </View>
  <View
    style={
      {
        "flexDirection": "column",
        "width": 40,
      }
    }
  />
</View>
`;
